File: Atividade4/src/Conta.java
public class Conta {
    private int idConta;
    private String numero;
    private double saldo;

    public Conta(int id, String numero, double saldo){
        this.idConta = id;
        this.numero = numero;
        this.saldo = saldo;
    }

    public boolean sacar(double valor){
        if (this.saldo >= valor){
            this.saldo -= valor;
            return true;
        }
        
        return false;
    }

    public void depositar(double valor){
        if(valor > 0){
            this.saldo += valor;
        }
    }

    public boolean transferir(Conta conta, double valor){
        if(this.sacar(valor)){
            conta.depositar(valor);
            return true;
        }

        return false;

    }

    public void verificarSaldo(){
        System.out.println("Saldo: R$ " + this.saldo);
    }

    public int getIdConta() {
        return this.idConta;
    }

    public void setIdConta(int idConta) {
        this.idConta = idConta;
    }

    public String getNumero() {
        return this.numero;
    }

    public void setNumero(String numero) {
        this.numero = numero;
    }

    public double getSaldo() {
        return this.saldo;
    }

    public void setSaldo(double saldo) {
        this.saldo = saldo;
    }

}

File: Atividade7-TS/src/sistema-bancario-ely/Conta.ts
import { Cliente } from "./Cliente";

export class Conta {
    private _id: number;
    private _numero: string;
    private _saldo: number;
    private _cliente: Cliente | null;
    private _dataDeAbertura: Date;

    constructor(numero: string, saldo: number) {
        this._id = 0;
        this._numero = numero;
        this._saldo = saldo;
        this._cliente = null;
        this._dataDeAbertura = new Date();
    }

    sacar(valor: number): void {
        this._saldo = this._saldo - valor;
    }

    depositar(valor: number): void {
        this._saldo = this._saldo + valor;
    }

    get saldo(): number {
        return this._saldo
    }


    transferir(contaDestino: Conta, valor: number): void {
        // this.saldo = this.saldo - valor;
        // contaDestino.saldo = contaDestino.saldo + valor;

        this.sacar(valor);
        contaDestino.depositar(valor);
    }

    get numero(): string {
        return this._numero;
    }

    get id(): number {
        return this._id;
    }

    set id(umId: number) {
        this._id = umId;
    }

    get cliente(): Cliente | null {
        return this._cliente;
    }

    set cliente(umCliente: Cliente | null) {
        this._cliente = umCliente;
    }
}
